src/abstract-interpretation/data-frame/mappers/assignment-mapper.ts

```diff
@@ -0,0 +1,41 @@
+import { VariableResolve } from '../../../config';
+import type { DataflowGraph } from '../../../dataflow/graph/graph';
+import type { RNode } from '../../../r-bridge/lang-4.x/ast/model/model';
+import type { RString } from '../../../r-bridge/lang-4.x/ast/model/nodes/r-string';
+import type { RSymbol } from '../../../r-bridge/lang-4.x/ast/model/nodes/r-symbol';
+import type { ParentInformation } from '../../../r-bridge/lang-4.x/ast/model/processing/decorate';
+import { RType } from '../../../r-bridge/lang-4.x/ast/model/type';
+import type { DataFrameAssignmentInfo } from '../absint-info';
+import { isDataFrameArgument } from './arguments';
+
+/**
+ * Maps a concrete data frame assignment to data frame assignment info containing the ids of the identifier and assigned expression.
+ *
+ * @param identifier - The R node of the variable identifier
+ * @param expression - The R node of the assigned expression
+ * @param dfg  - The data flow graph for resolving the arguments
+ * @returns Data frame assignment info containing the IDs of the identifier and expression, or `undefined` if the node does not represent a data frame assignment
+ */
+export function mapDataFrameVariableAssignment(
+	identifier: RSymbol<ParentInformation> | RString<ParentInformation>,
+	expression: RNode<ParentInformation>,
+	dfg: DataflowGraph
+): DataFrameAssignmentInfo | undefined {
+	const resolveInfo = { graph: dfg, idMap: dfg.idMap, full: true, resolve: VariableResolve.Alias };
+
+	if(!isDataFrameArgument(expression, resolveInfo)) {
```

Comment:
So this way we die in functions? this is fine but maybe we should document this

Reply:
no i mean we cannot detect a dataframe transformer function within a function definition but i think we should just document that that in general these only kick when we know it is a data frame